src/sage/schemes/elliptic_curves/ell_rational_field.py

```diff
@@ -2068,6 +2101,12 @@ def rank(self, use_database=True, verbose=False,
             0
             sage: E._EllipticCurve_rational_field__rank
             (0, True)
+
+            sage: E =EllipticCurve([-113^2,0])
+            sage: E.rank(use_database=False, verbose=False, algorithm="pari")
+            Traceback (most recent call last):
+            ...
+            RuntimeError: rank not provably correct (lower bound: 0, upper bound:2). Hint: increase pari_effort.
```

Comment:
Would a second run of this example with a larger pari effort give a different result?  If so, and it would not be too slow, you could add it.

Reply:
That is the example [-113^2,0]. Here Sha is Z/4 x Z/4 and so no matter how much effort we pout in we can't find points. I put the comment in.

Reply:
OK